### PR TITLE
Add flag for disabling flash attention

### PR DESCRIPTION
**Pull request by Maxusmusti**
For hardware scenarios where flash attention is potentially not supported.

### CODE REVIEW

**Comment by aldopareja** [comment]
are we sure that not specifying the attention mechanism is enough for this not to fail?. We need to setup the 3090 machine from Mo, I guess we can spinup docker containers there just to not mess up with her drivers?

File: main_ds.py [hunk 1]
```diff
@@ -57,21 +57,23 @@ def setup_model(args, tokenizer, train_loader, grad_accum):
...
+        base_model_args["attn_implementation"] = "flash_attention_2"
+
     if args.is_granite:
         from dolomite_engine.hf_models.models import GPTDolomiteForCausalLM
         model = GPTDolomiteForCausalLM.from_pretrained(
-            args.model_name_or_path,
-            attn_implementation="flash_attention_2",
-            torch_dtype=torch.bfloat16,
+            **base_model_args,
             use_padding_free_transformer=True,
```
Comment (fabianlim) [inline main_ds.py:72]: if you want to disable FA2, you also need to disable padding free, since it uses FA2 under the hood

**Review by RobotSail — COMMENTED** [review]
Will need to test this but otherwise it looks good.
